### PR TITLE
GH-41688: [Dev] Include all relevant CMakeLists.txt files in cmake-format precommit hook

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
### Rationale for this change

Some CMakeLists.txt files are not included in the pre-commit hook (causing failures on CI through archery if you rely on the pre-commit hook locally)

### What changes are included in this PR?

Include all CMakeLists.txt files by default anywhere in the repo, and explicitly exclude the ones we don't want (vendored files).

In practice, compared to the current set of files covered by the hook, those new files are included in the search:

'cpp/CMakeLists.txt',
'java/CMakeLists.txt',
'matlab/CMakeLists.txt',
'python/CMakeLists.txt'

### Are these changes tested?

Yes
* GitHub Issue: #41688

### CODE REVIEW

**Review by raulcd — APPROVED** [review]
Not an expert on those regex but the change LGTM

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -119,18 +119,15 @@ repos:
...
-          ?^python/.*/CMakeLists\.txt$|
+          ?.*CMakeLists\.txt$|
           )
         exclude: >-
           (
           ?^cpp/cmake_modules/FindNumPy\.cmake$|
           ?^cpp/cmake_modules/FindPythonLibsNew\.cmake$|
           ?^cpp/cmake_modules/UseCython\.cmake$|
           ?^cpp/src/arrow/util/config\.h\.cmake$|
+          ?^ci/conan/all/.*CMakeLists\.txt$|
```
Comment (kou) [inline .pre-commit-config.yaml:130]: Could you sort this list in alphabetical order?

```suggestion
          ?^ci/conan/all/.*CMakeLists\.txt$|
          ?^cpp/cmake_modules/FindNumPy\.cmake$|
          ?^cpp/cmake_modules/FindPythonLibsNew\.cmake$|
          ?^cpp/cmake_modules/UseCython\.cmake$|
          ?^cpp/src/arrow/util/config\.h\.cmake$|
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -119,18 +119,15 @@ repos:
           ?^ci/.*/.*\.cmake$|
           ?^cpp/.*/.*\.cmake\.in$|
           ?^cpp/.*/.*\.cmake$|
-          ?^cpp/.*/CMakeLists\.txt$|
-          ?^go/.*/CMakeLists\.txt$|
-          ?^java/.*/CMakeLists\.txt$|
-          ?^matlab/.*/CMakeLists\.txt$|
-          ?^python/.*/CMakeLists\.txt$|
+          ?.*CMakeLists\.txt$|
```
Comment (kou) [inline .pre-commit-config.yaml:122]: Could you sort this list in alphabetical order?

```text
          ?.*CMakeLists\.txt$|
          ?^ci/.*/.*\.cmake$|
          ?^cpp/.*/.*\.cmake\.in$|
          ?^cpp/.*/.*\.cmake$|
```

**Review by kou — APPROVED** [review]
+1

**Pull request merged**